packages/virtualized-lists/Lists/VirtualizedList.js

```diff
@@ -879,6 +969,10 @@ class VirtualizedList extends StateSafePureComponent<Props, State> {
           cellKey={this._getCellKey() + '-header'}
           key="$header">
           <View
+            // We expect that header component will be a single native view so make it
+            // not collapsable to avoid this view being flattened and make this assumption
+            // no longer true.
+            collapsable={false}
```

Comment:
View flattening of header component can causes issues here since we assume header view = 1 native view in minIndexForVisible. I think it is ok to always disable flattening here, but can also do it only if maintainVisibleContentPosition is true.

Reply:
This is also why the example had 2 as minIndexForVisible although it would be expected to work with only 1

Reply:
Not needed for now, but is the idea here that we need the native component alive around for `maintainVisibleContentPosition` to work?

Is it specifically that the anchor index needs to be visible? Wondering if this needs to live in ScrollView for that case when someone sets `minIndexForVisible`.

Reply:
It is more that minIndexForVisible index is based on native view index and not react native view index. So if we have a header view that gets flattened with 2 child view then our header now adds 2 child native views, so our adjustment of 1 view for header is now wrong.

We can’t really make this adjustment in ScrollView easily since collapsable prop needs to be set on ScrollView child views. It would def be possible but I don’t think it is needed.

Reply:
> So if we have a header view that gets flattened with 2 child view then our header now adds 2 child native views, so our adjustment of 1 view for header is now wrong.

Ah, thanks for the clarification!